core/templates/dev/head/services/context.service.ts

```diff
@@ -104,7 +105,24 @@ angular.module('oppia').factory('ContextService', [
           this.getPageContext() === PAGE_CONTEXT.EXPLORATION_PLAYER);
       },
 
+      // This function is used in cases where the URL does not specify the
+      // correct context for some case. eg: Viewing a skill's concept card on
+      // any page via the RTE.
+      setCustomEntity: function(entityType, entityId) {
```

Comment:
Should it be setCustomEntityContext?

Also please make an EntityContext domain object rather than using a raw dict. Then you can use entityContext.getId(), entityContext.getType().

Reply:
Done